test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +178,333 @@ int suiteTearDown( int numFailures )
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_TRUE( returnFlags.addAcknowledgment );
+    TEST_ASSERT_FALSE( returnFlags.endLoop );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Unsubscribe() works as intended.
+ */
+void test_MQTTAgentCommand_Unsubscribe( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentSubscribeArgs_t subscribeArgs = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_GetPacketId_ExpectAndReturn( &( mqttAgentContext.mqttContext ), 1 );
+    MQTT_Unsubscribe_ExpectAndReturn( &( mqttAgentContext.mqttContext ), subscribeArgs.pSubscribeInfo, subscribeArgs.numSubscriptions, 1, MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Unsubscribe( &mqttAgentContext, &subscribeArgs, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 1, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_TRUE( returnFlags.addAcknowledgment );
+    TEST_ASSERT_FALSE( returnFlags.endLoop );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Disconnect() works as intended.
+ */
+void test_MQTTAgentCommand_Disconnect( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Disconnect_ExpectAndReturn( &( mqttAgentContext.mqttContext ), MQTTSuccess );
+    mqttStatus = MQTTAgentCommand_Disconnect( &mqttAgentContext, NULL, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 0, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.endLoop );
+    TEST_ASSERT_FALSE( returnFlags.addAcknowledgment );
+    TEST_ASSERT_FALSE( returnFlags.runProcessLoop );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Ping() works as intended.
+ */
+void test_MQTTAgentCommand_Ping( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Ping_ExpectAndReturn( &( mqttAgentContext.mqttContext ), MQTTSuccess );
+    mqttStatus = MQTTAgentCommand_Ping( &mqttAgentContext, NULL, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 0, returnFlags.packetId );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_FALSE( returnFlags.addAcknowledgment );
+    TEST_ASSERT_FALSE( returnFlags.endLoop );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Connect() success case.
+ */
+void test_MQTTAgentCommand_Connect( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentConnectArgs_t connectInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Connect_IgnoreAndReturn( MQTTSuccess );
```

Comment:
Just do ExpectAnyArgsAndReturn instead of ignoring it
```suggestion
    MQTT_Connect_ExpectAnyArgsAndReturn( MQTTSuccess );
```